---
title: Multiqc v0.3.2
date: 2016-02-08
tags: [multiqc]
---

New version - a number of new modules, a few new command line options and some new toys in the report user interface.

I've just noticed that this was a bugfix version number bump and it should have been a minor version bump (`v0.4`). But I've already pushed this to PyPI so I'm not going to change it now. Sorry.

Anyway, here's what's new:

## Changelog
- All modules now load their log file search parameters from a config
  file, allowing you to overwrite them using your user config file
  - This is useful if your analysis pipeline renames program outputs
- New Picard (sub)modules - Insert Size, GC Bias & HsMetrics
- New Qualimap (sub)module - RNA-Seq QC
- Made Picard MarkDups show percent by default instead of counts
- Added M-Bias plot to Bismark
- New option to stream report HTML to `stdout`
- Files can now be specified as well as directories
- New options to specify whether the parsed data directory should be created
  - command line flags: `--data` / `--no-data`
  - config option name: `make_data_dir`
- Fixed bug with incorrect path to installation dir config YAML file
- New toolbox drawer for bulk-exporting graph images
- Report side navigation can now be hidden to maximise horizontal space
- Mobile styling improved for narrow screen
- More vibrant colours in the general stats table
- General stats table numbers now left aligned
- Settings now saved and loaded to named localstorage locations
  - Simplified interface - no longer global / single report saving
  - Removed static file config. Solves JS error, no-one was doing this
    since we have standalone reports anyway.
- Added support for Python 3.5
- Fixed bug with module specific CSS / JS includes in some templates
- Made the 'ignore files' config use unix style file pattern matching
- Fixed some bugs in the FastQ Screen module
- Fixed some bugs in the FastQC module
- Fixed occasional general stats table bug
- Table sorting on sample names now works after renaming
- Bismark module restructure
  - Each report type now handled independently (alignment / dedup / meth extraction)
  - M-Bias plot now shows R1 and R2
- FastQC GC content plot now has option for counts or percentages
  - Allows comparison between samples with very different read counts
- Bugfix for reports javascript
  - Caused by updated to remotely loaded HighCharts export script
  - Export script now bundled with multiqc, so does not depend on internet connection
  - Other JS errors fixed in this work
- Bugfix for older FastQC reports - handle old style sequence dup data
- Bugfix for varying Tophat alignment report formats
- Bugfix for Qualimap RNA Seq reports with paired end data
