---
title: MultiQC v1.3
date: 2017-11-03
tags: [multiqc]
---

There are 34 merged pull-requests in this release - a fantastic example of how an open source community can develop a tool! Many thanks to everyone involved for their hard work.

#### Breaking changes - custom search patterns
Only for users with custom search patterns for the `bowtie` or `star`: you will
need to update your config files - the `bowtie` search key is now `bowtie1`,
`star_genecounts` is now `star/genecounts`.

For users with custom modules - search patterns _must_ now conform to the search
pattern naming convention: `modulename` or `modulename/anything` (the search pattern
string beginning with the name of your module, anything you like after the first `/`).

#### New Modules:
* [**10X Supernova**](https://support.10xgenomics.com/de-novo-assembly/software/overview/welcome)
    * Parses statistics from the _de-novo_ Supernova software.
    * Module written by [@remiolsen](https://github.com/remiolsen/)
* [**BBMap**](https://sourceforge.net/projects/bbmap/)
    * Plot metrics from a number of BBMap tools, a suite of DNA/RNA mapping tools and utilities
    * Module written by [@boulund](https://github.com/boulund/) and [@epruesse](https://github.com/epruesse/)
* [**deepTools**](https://github.com/fidelram/deepTools) - new module!
    * Parse text output from `bamPEFragmentSize`, `estimateReadFiltering`, `plotCoverage`, `plotEnrichment`, and `plotFingerprint`
    * Module written by [@dpryan79](https://github.com/dpryan79/)
* [**Homer Tag Directory**](http://homer.ucsd.edu/homer/ngs/tagDir.html) - new submodule!
    * Module written by [@rdali](https://github.com/rdali/)
* [**illumina InterOp**](http://illumina.github.io/interop/index.html)
    * Module to parse metrics from illumina sequencing runs and demultiplexing, generated by the InterOp package
    * Module written by [@matthdsm](https://github.com/matthdsm/)
* [**RSEM**](https://deweylab.github.io/RSEM/) - new module!
    * Parse `.cnt` file comming from rsem-calculate-expression and plot read repartitions (Unalignable, Unique, Multi ...)
    * Module written by [@noirot](https://github.com/noirot/)
* [**HiCExplorer**](https://github.com/maxplanck-ie/HiCExplorer)
    * New module to parse the log files of `hicBuildMatrix`.
    * Module written by [@joachimwolff](https://github.com/joachimwolff/)

#### Module updates:
* **AfterQC**
    * Handle new output format where JSON summary key changed names.
* **bcl2fastq**
    * Clusters per sample plot now has tab where counts are categoried by lane.
* **GATK**
    * New submodule to handle Base Recalibrator stats, written by [@winni2k](https://github.com/winni2k/)
* **HiSAT2**
    * Fixed bug where plot title was incorrect if both SE and PE bargraphs were in one report
* **Picard HsMetrics**
    * Parsing code can now handle commas for decimal places
* **Preseq**
    * Updated odd file-search pattern that limited input files to 500kb
* **QoRTs**
    * Added new plots, new helptext and updated the module to produce a lot more output.
* **Qualimap BamQC**
    * Fixed edge-case bug where the refactored coverage plot code could raise an error from the `range` call.
* Documentation and link fixes for Slamdunk, GATK, bcl2fastq, Adapter Removal, FastQC and main docs
    * Many of these spotted and fixed by [@juliangehring](https://github.com/juliangehring/)
* Went through all modules and standardised plot titles
    * All plots should now have a title with the format _Module name: Plot name_

#### New MultiQC Features:
* New MultiQC docker image
    * Ready to use docker image now available at https://hub.docker.com/r/ewels/multiqc/ (200 MB)
    * Uses automated builds - pull `:latest` to get the development version, future releases will have stable tags.
    * Written by [@MaxUlysse](https://github.com/MaxUlysse/)
* New `module_order` config options allow modules to be run multiple times
    * Filters mean that a module can be run twice with different sets of files (eg. before and after trimming)
    * Custom module config parameters can be passed to module for each run
* File search refactored to only search for running modules
    * Makes search much faster when running with lots of files and limited modules
    * For example, if using `-m star` to only use the STAR module, all other file searches now skipped
* File search now warns if an unrecognised search type is given
* MultiQC now saves nearly all parsed data to a structured output file by default
    * See `multiqc_data/multiqc_data.json`
    * This can be turned off by setting `config.data_dump_file: false`
* Verbose logging when no log files found standardised. Less duplication in code and logs easier to read!
* New documentation section describing how to use MultiQC with Galaxy
* Using `shared_key: 'read_counts'` in table header configs now applies relevant defaults

#### Bug Fixes
* Installation problem caused by changes in upstream dependencies solved by stricter installation requirements
* Minor `default_dev` directory creation bug squashed
* Don't prepend the directory separator (`|`) to sample names with `-d` when there are no subdirs
* `yPlotLines` now works even if you don't set `width`
