---
title: MultiQC v1.5
date: 2018-03-15
tags: [multiqc]
---


#### New Modules:
* [**DeDup**](http://www.github.com/apeltzer/DeDup) - New module!
    * DeDup: Improved Duplicate Removal for merged/collapsed reads in ancient DNA analysis
    * Module written by [@apeltzer](https://github.com/apeltzer),
* [**Clip&Merge**](http://github.com/apeltzer/ClipAndMerge) - New module!
    * Clip&Merge: Adapter clipping and read merging for ancient DNA analysis
    * Module written by [@apeltzer](https://github.com/apeltzer),

#### Module updates:
* **bcl2fastq**
    * Catch `ZeroDivisionError` exceptions when there are 0 reads ([@aledj2](https://github.com/aledj2))
    * Add parsing of `TrimmedBases` and new General Stats column for % bases trimmed ([@matthdsm](https://github.com/matthdsm)).
* **BUSCO**
    * Fixed configuration bug that made all sample names become `'short'`
* **Custom Content**
    * Parsed tables now exported to `multiqc_data` files
* **Cutadapt**
    * Refactor parsing code to collect all length trimming plots
* **FastQC**
    * Fixed starting y-axis label for GC-content lineplot being incorrect.
* **HiCExplorer**
    * Updated to work with v2.0 release.
* **Homer**
    * Made parsing of `tagInfo.txt` file more resilient to variations in file format so that it works with new versions of Homer.
    * Kept order of chromosomes in coverage plot consistent.
* **Peddy**
    * Switch `Sex error` logic to `Correct sex` for better highlighting ([@aledj2](https://github.com/aledj2))
* **Picard**
    * Updated module and search patterns to recognise new output format from Picard version >= 2.16 and GATK output.
* **Qualimap BamQC**
    * Fixed bug where start of _Genome Fraction_ could have a step if target is 100% covered.
* **RNA-SeQC**
    * Added rRNA alignment stats to summary table [@Rolandde](https://github.com/Rolandde)
* **RSeqC**
    * Fixed read distribution plot by adding category for `other_intergenic` (thanks to [@moxgreen](https://github.com/moxgreen))
    * Fixed a dodgy plot title (Read GC content)
* **Supernova**
    * Added support for Supernova 2.0 reports. Fixed a TypeError bug when using txt reports only. Also a bug when parsing empty histogram files.

#### New MultiQC Features:
* Invalid choices for `--module` or `--exclude` now list the available modules alphabetically.
* Linting now checks for presence in `config.module_order` and tags.

#### Bug Fixes
* Excluding modules now works in combination with using module tags.
* Fixed edge-case bug where certain combinations of `output_fn_name` and `data_dir_name` could trigger a crash
* Conditional formatting - values are now longer double-labelled
* Made config option `extra_series` work in scatter plots the same way that it works for line plots
* Locked the `matplotlib` version to `v2.1.0` and below
    * Due to [two](https://github.com/matplotlib/matplotlib/issues/10476) [bugs](https://github.com/matplotlib/matplotlib/issues/10784) that appeared in `v2.2.0` - will remove this constraint when there's a new release that works again.
