---
title: MultiQC v1.6
date: 2018-08-04
tags: [multiqc]
---

Some of these updates are thanks to the efforts of people who attended the [NASPM](https://twitter.com/NordicGenomics) 2018 MultiQC hackathon session. Thanks to everyone who attended!

#### New Modules:
* [**fastp**](https://github.com/OpenGene/fastp)
    * An ultra-fast all-in-one FASTQ preprocessor (QC, adapters, trimming, filtering, splitting...)
    * Module started by [@florianduclot](https://github.com/florianduclot/) and completed by [@ewels](https://github.com/ewels/)
* [**hap.py**](https://github.com/Illumina/hap.py)
    * Hap.py is a set of programs based on htslib to benchmark variant calls against gold standard truth datasets
    * Module written by [@tsnowlan](https://github.com/tsnowlan/)
* [**Long Ranger**](https://support.10xgenomics.com/genome-exome/software/pipelines/latest/what-is-long-ranger)
    * Works with data from the 10X Genomics Chromium. Performs sample demultiplexing, barcode processing, alignment, quality control, variant calling, phasing, and structural variant calling.
    * Module written by [@remiolsen](https://github.com/remiolsen/)
* `**miRTrace**`
    * A quality control software for small RNA sequencing data.
    * Module written by [@chuan-wang](https://github.com/chuan-wang/)

#### Module updates:
* **BCFtools**
    * New plot showing SNP statistics versus quality of call from bcftools stats ([@MaxUlysse](https://github.com/MaxUlysse) and [@Rotholandus](https://github.com/Rotholandus))
* **BBMap**
    * Support added for BBDuk kmer-based adapter/contaminant filtering summary stats ([@boulund](https://github.com/boulund)
* **FastQC**
    * New read count plot, split into unique and duplicate reads if possible.
    * Help text added for all sections, mostly copied from the excellent FastQC help.
    * Sequence duplication plot rescaled
* **FastQ Screen**
    * Samples in large-sample-number plot are now sorted alphabetically ([@hassanfa](https://github.com/hassanfa)
* **MACS2**
    * Output is now more tolerant of missing data (no plot if no data)
* **Peddy**
    * Background samples now shown in ancestry PCA plot ([@roryk](https://github.com/roryk))
    * New plot showing sex checks versus het ratios, supporting unknowns ([@oyvinev](https://github.com/oyvinev))
* **Picard**
    * New submodule to handle `ValidateSamFile` reports ([@cpavanrun](https://github.com/cpavanrun))
    * WGSMetrics now add the mean and standard-deviation coverage to the general stats table (hidden) ([@cpavanrun](https://github.com/cpavanrun))
* **Preseq**
    * New config option to plot preseq plots with unique old coverage on the y axis instead of read count
    * Code refactoring by [@vladsaveliev](https://github.com/vladsaveliev)
* **QUAST**
    * Null values (`-`) in reports now handled properly. Bargraphs always shown despite varying thresholds. ([@vladsaveliev](https://github.com/vladsaveliev))
* **RNA-SeQC**
    * Don't create the report section for Gene Body Coverage if no data is given
* **Samtools**
    * Fixed edge case bug where MultiQC could crash if a sample had zero count coverage with idxstats.
    * Adds % proper pairs to general stats table
* **Skewer**
    * Read length plot rescaled
* **Tophat**
    * Fixed bug where some samples could be given a blank sample name ([@lparsons](https://github.com/lparsons))
* **VerifyBamID**
    * Change column header help text for contamination to match percentage output ([@chapmanb](https://github.com/chapmanb))

#### New MultiQC Features:
* New config option `remove_sections` to skip specific report sections from modules
* Add `path_filters_exclude` to exclude certain files when running modules multiple times. You could previously only include certain files.
* New `exclude_*` keys for file search patterns
    * Have a subset of patterns to exclude otherwise detected files with, by filename or contents
* Command line options all now use mid-word hyphens (not a mix of hyphens and underscores)
    * Old underscore terms still maintained for backwards compatibility
* Flag `--view-tags` now works without requiring an "analysis directory".
* Removed Python dependency for `enum34` ([@boulund](https://github.com/boulund))
* Columns can be added to `General Stats` table for custom content/module.
* New `--ignore-symlinks` flag which will ignore symlinked directories and files.
* New `--no-megaqc-upload` flag which disables automatically uploading data to MegaQC

#### Bug Fixes
* Fix path_filters for top_modules/module_order configuration only selecting if *all* globs match. It now filters searches that match *any* glob.
* Empty sample names from cleaning are now no longer allowed
* Stop prepend_dirs set in the config from getting clobbered by an unpassed CLI option ([@tsnowlan](https://github.com/tsnowlan))
* Modules running multiple times now have multiple sets of columns in the General Statistics table again, instead of overwriting one another.
* Prevent tables from clobbering sorted row orders.
* Fix linegraph and scatter plots data conversion (sporadically the incorrect `ymax` was used to drop data points) ([@cpavanrun](https://github.com/cpavanrun))
* Adjusted behavior of ceiling and floor axis limits
* Adjusted multiple file search patterns to make them more specific
    * Prevents the wrong module from accidentally slurping up output from a different tool. By [@cpavanrun](https://github.com/cpavanrun) (see [PR #727](https://github.com/ewels/MultiQC/pull/727))
* Fixed broken report bar plots when `-p`/`--export-plots` was specified (see issue [#801](https://github.com/ewels/MultiQC/issues/801))
