---
title: Multiqc v1.7
date: 2018-12-21
tags: [multiqc]
---

An early Christmas present for MultiQC users! ðŸŽ…ðŸŽðŸŽ„

Many thanks to everyone who has contributed to this release. Happy Christmas and a very happy new year!

#### New Modules:
* [**BISCUIT**](https://github.com/zwdzwd/biscuit)
    * BISuilfite-seq CUI Toolkit
    * Module written by [@zwdzwd](https://github.com/zwdzwd/)
* [**DamageProfiler**](https://github.com/Integrative-Transcriptomics/DamageProfiler)
    * A tool to determine ancient DNA misincorporation rates.
    * Module written by [@apeltzer](https://github.com/apeltzer/)
* [**FLASh**](https://ccb.jhu.edu/software/FLASH/)
    * FLASH (Fast Length Adjustment of SHort reads)
    * Module written by [@pooranis](https://github.com/pooranis/)
* [**MinIONQC**](https://github.com/roblanf/minion_qc)
    * QC of reads from ONT long-read sequencing
    * Module written by [@ManavalanG](https://github.com/ManavalanG)
* [**phantompeakqualtools**](https://www.encodeproject.org/software/phantompeakqualtools)
    * A tool for informative enrichment and quality measures for ChIP-seq/DNase-seq/FAIRE-seq/MNase-seq data.
    * Module written by [@chuan-wang](https://github.com/chuan-wang/)
* [**Stacks**](http://catchenlab.life.illinois.edu/stacks/)
    * A software for analyzing restriction enzyme-based data (e.g. RAD-seq). Support for Stacks >= 2.1 only.
    * Module written by [@remiolsen](https://github.com/remiolsen/)

#### Module updates:
* **AdapterRemoval**
    * Handle error when zero bases are trimmed. See [#838](https://github.com/ewels/MultiQC/issues/838).
* **Bcl2fastq**
    * New plot showing the top twenty of undetermined barcodes by lane.
    * Informations for R1/R2 are now separated in the General Statistics table.
    * SampleID is concatenate with SampleName because in Chromium experiments several sample have the same SampleName.
* **deepTools**
    * New PCA plots from the `plotPCA` function (written by [@chuan-wang](https://github.com/chuan-wang/))
    * New fragment size distribution plots from `bamPEFragmentSize --outRawFragmentLengths` (written by [@chuan-wang](https://github.com/chuan-wang/))
    * New correlation heatmaps from the `plotCorrelation` function (written by [@chuan-wang](https://github.com/chuan-wang/))
    * New sequence distribution profiles around genes, from the `plotProfile` function (written by [@chuan-wang](https://github.com/chuan-wang/))
    * Reordered sections
* **Fastp**
    * Fixed bug in parsing of empty histogram data. See [#845](https://github.com/ewels/MultiQC/issues/845).
* **FastQC**
    * Refactored _Per Base Sequence Content_ plots to show original underlying data, instead of calculating it from the page contents. Now shows original FastQC base-ranges and fixes 100% GC bug in final few pixels. See [#812](https://github.com/ewels/MultiQC/issues/812).
    * When including a FastQC section multiple times in one report, the summary progress bars now behave as you would expect.
* **FastQ Screen**
    * Don't hide genomes in the simple plot, even if they have zero unique hits. See [#829](https://github.com/ewels/MultiQC/issues/829).
* **InterOp**
    * Fixed bug where read counts and base pair yields were not displaying in tables correctly.
    * Number formatting for these fields can now be customised in the same way as with other modules, as described [in the docs](http://multiqc.info/docs/#number-base-multiplier)
* **Picard**
    * InsertSizeMetrics: You can now configure to what degree the insert size plot should be smoothed.
    * CollectRnaSeqMetrics: Add warning about missing rRNA annotation.
    * CollectRnaSeqMetrics: Add chart for counts/percentage of reads mapped to the correct strand.
    * Now parses VariantCallingMetrics reports. (Similar to GATK module's VariantEval.)
* **phantompeakqualtools**
    * Properly clean sample names
* **Trimmomatic**
    * Updated Trimmomatic module documentation to be more helpful
    * New option to use filenames instead of relying on the command line used. See [#864](https://github.com/ewels/MultiQC/issues/864).

#### New MultiQC Features:
* Embed your custom images with a new Custom Content feature! Just add `_mqc` to the end of the filename for `.png`, `.jpg` or `.jpeg` files.
* Documentation for Custom Content reordered to make it a little more sane
* You can now add or override any config parameter for any MultiQC plot! See [the documentation](http://multiqc.info/docs/#customising-plots) for more info.
* Allow `table_columns_placement` config to work with table IDs as well as column namespaces. See [#841](https://github.com/ewels/MultiQC/issues/841).
* Improved visual spacing between grouped bar plots


#### Bug Fixes:
* Custom content no longer clobbers `col1_header` table configs
* The option `--file-list` that refers to a text file with file paths to analyse will no longer ignore directory paths
* [Sample name directory prefixes](https://multiqc.info/docs/#sample-names-prefixed-with-directories) are now added _after_ cleanup.
* If a module is run multiple times in one report, it's CSS and JS files will only be included once (`default` template)
